tests/test_sample_size/test_sample_size_calculator.py

```diff
@@ -0,0 +1,216 @@
+        self.assertEqual(calculator.power, DEFAULT_POWER)
+        self.assertEqual(calculator.boolean_metrics, [])
+        self.assertEqual(calculator.numeric_metrics, [])
+        self.assertEqual(calculator.ratio_metrics, [])
+
+    def test_register_bool_metric(self):
+        test_probability = 0.05
+        test_mde = 0.02
+
+        calculator = SampleSizeCalculator()
+        calculator.register_bool_metric(test_probability, test_mde)
+
+        self.assertEqual(len(calculator.boolean_metrics), 1)
+        self.assertEqual(calculator.boolean_metrics[0].variance, 0.0475)
+        self.assertEqual(calculator.boolean_metrics[0].mde, test_mde)
+        self.assertEqual(calculator.numeric_metrics, [])
+        self.assertEqual(calculator.ratio_metrics, [])
+
+        calculator.register_bool_metric(test_probability, test_mde)
+        self.assertEqual(len(calculator.boolean_metrics), 2)
+
+    def test_register_numeric_metric(self):
+        test_variance = 5000
+        test_mde = 10
+
+        calculator = SampleSizeCalculator()
+        calculator.register_numeric_metric(test_variance, test_mde)
+
+        self.assertEqual(len(calculator.numeric_metrics), 1)
+        self.assertEqual(calculator.numeric_metrics[0].variance, test_variance)
+        self.assertEqual(calculator.numeric_metrics[0].mde, test_mde)
+        self.assertEqual(calculator.boolean_metrics, [])
+        self.assertEqual(calculator.ratio_metrics, [])
+
+        calculator.register_numeric_metric(test_variance, test_mde)
+        self.assertEqual(len(calculator.numeric_metrics), 2)
+
+    def test_register_ratio_metric(self):
+        test_numerator_mean = 2000
+        test_numerator_variance = 100000
+        test_denominator_mean = 200
+        test_denominator_variance = 2000
+        test_covariance = 5000
+        test_mde = 10
+        test_variance = 5
+
+        calculator = SampleSizeCalculator()
+        calculator.register_ratio_metric(
+            test_numerator_mean,
+            test_numerator_variance,
+            test_denominator_mean,
+            test_denominator_variance,
+            test_covariance,
+            test_mde,
+        )
+
+        self.assertEqual(len(calculator.ratio_metrics), 1)
+        self.assertEqual(calculator.ratio_metrics[0].variance, test_variance)
+        self.assertEqual(calculator.ratio_metrics[0].mde, test_mde)
+        self.assertEqual(calculator.boolean_metrics, [])
+        self.assertEqual(calculator.numeric_metrics, [])
+
+        calculator.register_ratio_metric(
+            test_numerator_mean,
+            test_numerator_variance,
+            test_denominator_mean,
+            test_denominator_variance,
+            test_covariance,
+            test_mde,
+        )
+        self.assertEqual(len(calculator.ratio_metrics), 2)
+
+    @patch("statsmodels.stats.power.NormalIndPower.solve_power")
+    def test_get_single_sample_size_normal(self, mock_solve_power):
```

Comment:
Do we have any appetite for parameterizing this test?

```python3
@parameterized([
    {"metric": BooleanMetric(...), "solver": NormalIndPower},
    {"metric": NumericMetric(...), "solver": TTestIndPower}
])
@patch("statsmodels.stats.power.")
def test_get_single_sample_size_normal(self, mock_power, metric, solver):
    ...
```

Reply:
We could do this if we have more types in the future.